cli/alert.go

```diff
@@ -0,0 +1,189 @@
+  of "Matcher Groups" that will be used to filter your query. The following
+  examples will attempt to show this behaviour in action:
+
+  amtool alert query alertname=foo node=bar
+
+  	This query will match all alerts with the alertname=foo and node=bar label
+  	value pairs set.
+
+  amtool alert query foo node=bar
+
+  	If alertname is ommited and the first argument does not contain a '=' or a
+  	'=~' then it will be assumed to be the value of the alertname pair.
+
+  amtool alert query 'alertname=~foo.*'
+
+  	As well as direct equality, regex matching is also supported. The '=~' syntax
+  	(similar to prometheus) is used to represent a regex match. Regex matching
+  	can be used in combination with a direct match.
+
+  amtool alert query alertname=foo node={bar,baz}
+
+  	This query will match all alerts with the alertname=foo label value pair
+  	and EITHER node=bar or node=baz.
+
+  amtool alert query alertname=foo{a,b} node={bar,baz}
+
+	Similar to the previous example this query will match all alerts with any
+	combination of alertname=fooa or alertname=foob AND node=bar or node=baz.
+
+	`,
+	RunE: queryAlerts,
+}
+
+func init() {
+	RootCmd.AddCommand(alertCmd)
+	alertCmd.Flags().Bool("expired", false, "Show expired alerts as well as active")
+	alertCmd.Flags().BoolP("silenced", "s", false, "Show silenced alerts")
+	alertFlags = alertCmd.Flags()
+}
+
+func fetchAlerts(filter *[]labels.Matcher) (model.Alerts, error) {
+	alertResponse := alertmanagerAlertResponse{}
+
+	u, err := GetAlertmanagerURL()
+	if err != nil {
+		return model.Alerts{}, err
+	}
+
+	u.Path = path.Join(u.Path, "/api/v1/alerts")
+	if filter != nil {
+		u.RawQuery = url.QueryEscape(MatchersToString(*filter))
+	}
+
+	fmt.Println(u.String())
+
+	res, err := http.Get(u.String())
+	if err != nil {
+		return model.Alerts{}, err
+	}
+
+	defer res.Body.Close()
+	decoder := json.NewDecoder(res.Body)
+
+	err = decoder.Decode(&alertResponse)
+	if err != nil {
+		return model.Alerts{}, errors.New("Unable to decode json response")
+	}
+	return alertResponse.Data, nil
+}
+
+func queryAlerts(cmd *cobra.Command, args []string) error {
+	silences, err := fetchSilences(nil)
+	if err != nil {
+		return err
+	}
+
+	var groups [][]labels.Matcher
+	if len(args) > 0 {
+		matchers, err := parseMatchers(args)
+		if err != nil {
+			return err
+		}
+		groups = parseMatcherGroups(matchers)
+		if err != nil {
+			return err
+		}
+	}
+
+	expired, err := alertFlags.GetBool("expired")
+	if err != nil {
+		return err
+	}
+
+	showSilenced, err := alertFlags.GetBool("silenced")
+	if err != nil {
+		return err
+	}
+
+	fetchedAlerts := model.Alerts{}
+	if len(groups) < 1 {
+		fmt.Println("No specified matchers")
+		fetchedAlerts, err = fetchAlerts(nil)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Fetch all alerts that match each group of matchers
+	for _, matchers := range groups {
+		alerts, err := fetchAlerts(&matchers)
```

Comment:
I think having a single query that gets sent will end up being easier to reason about